### PR TITLE
nrf: doc: release notes for TF-M and for Zephyr "architectures" area

### PR DESCRIPTION
**Pull request by ioannisg**
@oyvindronningstad pls revisit the added content about TF-M.

### CODE REVIEW

**Review by oyvindronningstad — APPROVED** [review]

File: doc/nrf/releases/release-notes-latest.rst [hunk 1]
```diff
@@ -233,6 +233,10 @@ Trusted Firmware-M:
 -------------------
 
 * Added a simple sample that demonstrates how to integrate TF-M in an application.
+* Made it possible to use platform code residing outside of the trusted-firmware-m
```
Comment (b-gent) [inline doc/nrf/releases/release-notes-latest.rst:236]:
```suggestion
* Enabled using the platform code that resides outside of the Trusted Firmware-M repository.
  This allows for providing configurable memory partitioning in the nRF Connect SDK.
```

**Review by b-gent — CHANGES_REQUESTED** [review]

**Review by oyvindronningstad — APPROVED** [review]

File: doc/nrf/releases/release-notes-latest.rst (hunk 1 above)
Comment (ioannisg) [reply doc/nrf/releases/release-notes-latest.rst]: done @b-gent , pls, revisit.

File: doc/nrf/releases/release-notes-latest.rst [hunk 2]
```diff
@@ -250,6 +250,10 @@ Trusted Firmware-M:
 -------------------
 
 * Added a simple sample that demonstrates how to integrate TF-M in an application.
+* Enabled using the platform code that resides outside of the Trusted Firmware-M
```
Comment (oyvindronningstad) [inline doc/nrf/releases/release-notes-latest.rst:253]:
```suggestion
* Enabled using platform code that resides outside of the Trusted Firmware-M
```

**Review by oyvindronningstad — CHANGES_REQUESTED** [review]

File: doc/nrf/releases/release-notes-latest.rst [hunk 3]
```diff
@@ -250,6 +250,10 @@ Trusted Firmware-M:
...
+  repository. This allows for providing configurable memory partitioning in the
```
Comment (b-gent) [inline doc/nrf/releases/release-notes-latest.rst:254]: @ioannisg please change to adhere to 'one sentence - one line' rule in RST docs ( no 100 char limit here)
So the first sentence should not be broken and the second one should start in newline.

File: doc/nrf/releases/release-notes-latest.rst (hunk 2 above)
Comment (ioannisg) [reply doc/nrf/releases/release-notes-latest.rst]: done @oyvindronningstad thanks

**Review by oyvindronningstad — APPROVED** [review]

**Review by b-gent — APPROVED** [review]

**Pull request merged**